service/routes.py

```diff
@@ -206,44 +206,50 @@ def get_shopcart_items(shopcart_id):
 ######################################################################
 # G E T  S H O P C A R T  I T E M
 ######################################################################
-@app.route("/shopcartitem/<shopcart_item_id>", methods=["GET"])
-def get_shopcart_item(shopcart_item_id):
+@app.route('/shopcarts/<int:shopcart_id>/items/<int:item_id>', methods=["GET"])
+def get_shopcart_item(shopcart_id, item_id):
     """
     Get a shopcart item
     This endpoint will return an item in the shop cart
     """
     logger.info("Request to get an item in a shopcart")
 
-    shopcart_item =  ShopcartItem.find(shopcart_item_id)
+    if item_id != shopcart_id:
+        raise BadRequest("Shopcart id '{}' does not match item id '{}'.".format(
+                         shopcart_id, item_id))
+
+    shopcart_item =  ShopcartItem.find(item_id)
 
     if shopcart_item is None:
-        logger.info("Shopcart item with ID [%s] not found.", shopcart_item_id)
+        logger.info("Shopcart item with ID [%s] not found.", item_id)
         return not_found("Not found")
 
     result = shopcart_item.serialize()
-    logger.info("Fetched shopcart item with ID [%s].", shopcart_item_id)
+    logger.info("Fetched shopcart item with ID [%s].", item_id)
     return make_response(
         jsonify(result), status.HTTP_200_OK
     )
 
 ######################################################################
 # ADD A NEW SHOPCARTITEM
 ######################################################################
-@app.route("/shopcartitems", methods=["POST"])
-def create_shopcart_items():
+@app.route("/shopcarts/<int:shopcart_id>/items", methods=["POST"])
+def create_shopcart_items(shopcart_id):
     """
     Creates a ShopcartItem
     This endpoint will create a ShopcartItem based the data in the body that is posted
     """
     logger.info("Request to create a shopcart item")
     check_content_type("application/json")
     shopcart_item = ShopcartItem()
-    print(request.get_json)
     shopcart_item.deserialize(request.get_json())
+    if shopcart_item.sid != shopcart_id:
+        raise BadRequest("Shopcart id '{}' does not match sid in request '{}'.".format(
+                          shopcart_id, shopcart_item.sid))
```

Comment:
If `shopcart_id` is in the URL, do we really want `sid` in the request body?

Reply:
Good point, I changed it to set it to the id contained in the URL